requirements/views.py

```diff
@@ -139,11 +140,44 @@ def progress(request, list_id, courses):
 
     # Get Course objects
     course_objs = []
+    #required to give generic courses unique id's so muliple can count towards requirement
+    unique_generic_id = 0
     for subject_id in [c for c in courses.split(",") if len(c)]:
         try:
             course_objs.append(Course.public_courses().get(subject_id=subject_id))
         except ObjectDoesNotExist:
-            print("Warning: course {} does not exist in the catalog".format(subject_id))
+            is_generic_course = False
+            if "." not in subject_id:
+                #potential a generic course
```

Comment:
Can you move this into a `Course` static method, e.g. `Course.make_generic(subject_id, unique_id)`? This method could raise a `ValueError` if it's not a generic course, prompting the "does not exist in catalog" warning. I think that would clean up the `progress` method quite a bit.